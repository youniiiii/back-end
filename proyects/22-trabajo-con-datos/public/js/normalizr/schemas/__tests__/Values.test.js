// eslint-env jest
import { fromJS } from 'immutable';
import { denormalize, normalize, schema } from '../../';

describe(`${schema.Values.name} normalization`, () => {
  test('normalizes the values of an object with the given schema', () => {
    const cat = new schema.Entity('cats');
    const dog = new schema.Entity('dogs');
    const valuesSchema = new schema.Values(
      {
        dogs: dog,
        cats: cat,
      },
      (entity, key) => entity.type
    );

    expect(
      normalize(
        {
          fido: { id: 1, type: 'dogs' },
          fluffy: { id: 1, type: 'cats' },
        },
        valuesSchema
      )
    ).toMatchSnapshot();
  });

  test('can use a function to determine the schema when normalizing', () => {
    const cat = new schema.Entity('cats');
    const dog = new schema.Entity('dogs');
    const valuesSchema = new schema.Values(
      {
        dogs: dog,
        cats: cat,
      },
      (entity, key) => `${entity.type}s`
    );

    expect(
      normalize(
        {
          fido: { id: 1, type: 'dog' },
          fluffy: { id: 1, type: 'cat' },
          jim: { id: 2, type: 'lizard' },
        },
        valuesSchema
      )
    ).toMatchSnapshot();
  });

  test('filters out null and undefined values', () => {
    const cat = new schema.Entity('cats');
    const dog = new schema.Entity('dogs');
    const valuesSchema = new schema.Values(
      {
        dogs: dog,
        cats: cat,
      },
      (entity, key) => entity.type
    );

    expect(
      normalize(
        {
          fido: undefined,
          milo: null,
          fluffy: { id: 1, type: 'cats' },
        },
        valuesSchema
      )
    ).toMatchSnapshot();
  });
});

describe(`${schema.Values.name} denormalization`, () => {
  test('denormalizes the values of an object with the given schema', () => {
    const cat = new schema.Entity('cats');
    const dog = new schema.Entity('dogs');
    const valuesSchema = new schema.Values(
      {
        dogs: dog,
        cats: cat,
      },
      (entity, key) => entity.type
    );

    const entities = {
      cats: { 1: { id: 1, type: 'cats' } },
      dogs: { 1: { id: 1, type: 'dogs' } },
    };

    expect(
      denormalize(
        {
          fido: { id: 1, schema: 'dogs' },
          fluffy: { id: 1, schema: 'cats' },
        },
        valuesSchema,
        entities
      )
    ).toMatchSnapshot();

    expect(
      denormalize(
        {
          fido: { id: 1, schema: 'dogs' },
          fluffy: { id: 1, schema: 'cats' },
        },
        valuesSchema,
        fromJS(entities)
      )
    ).toMatchSnapshot();
  });
});
