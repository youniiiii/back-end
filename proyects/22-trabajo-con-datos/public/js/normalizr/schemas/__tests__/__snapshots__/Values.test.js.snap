// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValuesSchema denormalization denormalizes the values of an object with the given schema 1`] = `
Object {
  "fido": Object {
    "id": 1,
    "type": "dogs",
  },
  "fluffy": Object {
    "id": 1,
    "type": "cats",
  },
}
`;

exports[`ValuesSchema denormalization denormalizes the values of an object with the given schema 2`] = `
Object {
  "fido": Immutable.Map {
    "id": 1,
    "type": "dogs",
  },
  "fluffy": Immutable.Map {
    "id": 1,
    "type": "cats",
  },
}
`;

exports[`ValuesSchema normalization can use a function to determine the schema when normalizing 1`] = `
Object {
  "entities": Object {
    "cats": Object {
      "1": Object {
        "id": 1,
        "type": "cat",
      },
    },
    "dogs": Object {
      "1": Object {
        "id": 1,
        "type": "dog",
      },
    },
  },
  "result": Object {
    "fido": Object {
      "id": 1,
      "schema": "dogs",
    },
    "fluffy": Object {
      "id": 1,
      "schema": "cats",
    },
    "jim": Object {
      "id": 2,
      "type": "lizard",
    },
  },
}
`;

exports[`ValuesSchema normalization filters out null and undefined values 1`] = `
Object {
  "entities": Object {
    "cats": Object {
      "1": Object {
        "id": 1,
        "type": "cats",
      },
    },
  },
  "result": Object {
    "fluffy": Object {
      "id": 1,
      "schema": "cats",
    },
  },
}
`;

exports[`ValuesSchema normalization normalizes the values of an object with the given schema 1`] = `
Object {
  "entities": Object {
    "cats": Object {
      "1": Object {
        "id": 1,
        "type": "cats",
      },
    },
    "dogs": Object {
      "1": Object {
        "id": 1,
        "type": "dogs",
      },
    },
  },
  "result": Object {
    "fido": Object {
      "id": 1,
      "schema": "dogs",
    },
    "fluffy": Object {
      "id": 1,
      "schema": "cats",
    },
  },
}
`;
